scripts/build-shrink-ray.js

```diff
@@ -1,73 +1,95 @@
-  execSync('yarn run storybook:build --quiet', {
-    stdio: 'inherit',
-  });
-}
+async function startShrinkRayBuild({
+  masterBranchName,
+  repo,
+  sha,
+  reportPath,
+  buildUrl,
+  skip,
+}) {
+  const logger = new Logger();
+
+  if (sha) {
+    logger.header('Running shrink-ray prebuild script...');
 
-function setupShrinkRay() {
-  console.log('[shrink-ray] Running shrink-ray prebuild script...');
+    // fetch latest in pipeline. Travis does a shallow clone by default,
+    // --unshallow makes sure we can fetch the master commit in case it is not
+    // included in the initial shallow clone
+    execSync('git fetch --unshallow origin master:refs/remotes/origin/master');
 
-  // fetch latest in pipeline. Travis does a shallow clone by default,
-  // --unshallow makes sure we can fetch the master commit in case it is not
-  // included in the initial shallow clone
-  execSync('git fetch --unshallow origin master:refs/remotes/origin/master');
+    const masterSha = execSync(
+      `git merge-base HEAD origin/${masterBranchName}`,
+      {
+        encoding: 'utf8',
+      },
+    ).trim();
 
-  const masterSha = execSync(`git merge-base HEAD origin/${BASE_BRANCH}`, {
-    encoding: 'utf8',
-  }).trim();
+    const shrinkRay = new ShrinkRayAPI();
+    const build = new Build({
+      repo,
+      sha,
+      masterSha,
+    });
 
-  sendCommitStatus({repo, sha, state: 'pending'})
-    .then(() => {
-      console.log('[shrink-ray] shrink-ray prebuild script completed.');
-      buildPackages();
+    try {
+      logger.header('Validating...');
+      build.validate();
+      logger.for('validation').log('✅ sha validation successful!');
 
-      return postReport({
-        repo,
-        sha,
-        baselineSha: masterSha,
-        reportPath: report,
-      });
-    })
-    .catch((error) => {
-      console.log('ERROR: ', error);
+      logger.header('Initializing check(s) on GitHub');
+      await shrinkRay.create({...build, buildUrl, skip});
 
-      // eslint-disable-next-line promise/no-nesting
-      sendCommitStatus({repo, sha, state: 'error'})
-        .then(() => {
-          throw error;
-        })
-        .catch(() => {
-          throw error;
-        });
-    });
+      buildStorybookBuild();
+
+      logger.header(`Uploading build report to Shrink-Ray`);
+      await shrinkRay.report(build, await bundleAnalyzerReport(reportPath));
+    } catch (error) {
+      logger.header('shrink-ray build fail');
+      logger.for('error').log(error);
+      logger.for('error').log(`sending status to server for clean up `);
+      await shrinkRay.error(build, error);
+    }
+  } else {
+    buildStorybookBuild();
+  }
+}
+
+function buildStorybookBuild() {
+  const logger = new Logger();
+
+  logger.header('Running storybook build...');
+  execSync('yarn run storybook:build --quiet', {
+    stdio: 'inherit',
+  });
+}
+
+function bundleAnalyzerReport(reportPath) {
+  try {
+    return readFile(reportPath, 'utf-8');
```

Comment:
This returns a promise which doesn't get resolved here - so I don't think the try/catch will ever do anything because nothing will ever throw before you resolve the promise? Change this to readFileSync perhaps?